pkg/metrics/rate.go

```diff
@@ -35,14 +35,14 @@ func (r *Rate) flush(timestamp float64) ([]*Serie, error) {
 	}
 
 	value, ts := (r.sample-r.previousSample)/(r.timestamp-r.previousTimestamp), r.timestamp
+	r.previousSample, r.previousTimestamp = r.sample, r.timestamp
+	r.sample, r.timestamp = 0., 0.
 
 	if value < 0 {
-		return []*Serie{}, fmt.Errorf("Rate value is negative (%f - %f = %f), discarding it (the underlying counter may have been reset)", r.sample, r.previousSample, value)
+		origSample, origPreviousSample := r.previousSample, value+r.previousSample
```

Comment:
I don't think `origPreviousSample` can be calculated simply by adding `previousSample` to `value`, remember that we've also divided by the time delta. It's not `(%f - %f = %f)` it's `((%f - %f) / delta) = %f` - so if you really want to report the sample, the log line won't be accurate.

Reply:
Yep, very true.  Will try again!